### ISSUE TITLE
Custom Iconset by iconfiy

### ISSUE THREAD

**Issue by cuiyajie** [selemondev/nuxt-ui-vue#143]
I notice that nuxt-ui-vue use iconify to support icon. That is good framework. Now I want to add my own icon collection, then use them in nuxt-ui-vue. 
```
import { addCollection } from '@iconify/vue'
addCollection({
    prefix: 's24',
    icons: tranIcons,
    width: 24,
    height: 24
  }, 'b3d')
``` 

for example
```
toast.add({  title: '1234',  timeout: 0, icon: '@b3d:s24:cross-filled' })
```

I find that `loadIcon` throw error in UIcon component. Any example can show how to use custom iconset?

**Comment by selemondev** [comment]
Yes, it does work. Here is an example:

```vue
<script setup lang='ts'>
import { addCollection } from '@iconify/vue';

addCollection({
	prefix: 'mdi',
	icons: {
		'account-box': {
			body: '<path d="M6 17c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6m9-9a3 3 0 0 1-3 3a3 3 0 0 1-3-3a3 3 0 0 1 3-3a3 3 0 0 1 3 3M3 5v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2z" fill="currentColor"/>',
		},
		'account-cash': {
			body: '<path d="M11 8c0 2.21-1.79 4-4 4s-4-1.79-4-4s1.79-4 4-4s4 1.79 4 4m0 6.72V20H0v-2c0-2.21 3.13-4 7-4c1.5 0 2.87.27 4 .72M24 20H13V3h11v17m-8-8.5a2.5 2.5 0 0 1 5 0a2.5 2.5 0 0 1-5 0M22 7a2 2 0 0 1-2-2h-3c0 1.11-.89 2-2 2v9a2 2 0 0 1 2 2h3c0-1.1.9-2 2-2V7z" fill="currentColor"/>',
		},
		'account': {
			body: '<path d="M12 4a4 4 0 0 1 4 4a4 4 0 0 1-4 4a4 4 0 0 1-4-4a4 4 0 0 1 4-4m0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4z" fill="currentColor"/>',
		},
		'home': {
			body: '<path d="M10 20v-6h4v6h5v-8h3L12 3L2 12h3v8h5z" fill="currentColor"/>',
		},
	},
	width: 24,
	height: 24,
});
</script>

<template>
  <div class="grid place-items-center w-full min-h-screen">
      <UIcon name="mdi:account-box" />
  </div>
</template>
```

**Comment by cuiyajie** [comment]
Yes, I test it and work well. But I want to use this with UNotification icon property. And it not work.

**Comment by cuiyajie** [comment]
I have tested your demo, if I change the prefix or icons body, the icon show incorrectly. It seems like the UIcon loads svg from iconify library not from custom defination.

**Comment by cuiyajie** [comment]
I check the code of UIcon in nuxt-ui-vue `import { Icon } from '@iconify/vue/dist/offline'`, I think we should use `import { Icon } from '@iconify/vue'` for custom iconset.

**Comment by selemondev** [comment]
Can you handle this by creating a PR?

**Comment by cuiyajie** [comment]
I find `@iconfiy/vue` create a globa variable `dataStorage` to store the custom icons. If I call `addCollections` from `@iconfiy/vue` in my code, it will create another `dataStorage` which is not shared with previous one in `nuxt-ui-vue`. we should expose the functions from `@iconfiy/vue`.

**Comment by cuiyajie** [comment]
PR  https://github.com/selemondev/nuxt-ui-vue/pull/145

**Comment by selemondev** [comment]
Thanks for your contribution and this issue has been successfully resolved with the latest release of Nuxt UI Vue v1.0.1-beta.4
